packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -540,20 +540,21 @@ class _AndroidMotionEventConverter {
     );
   }
 
-  void handlePointerUpEvent(PointerUpEvent event) {
-    pointerPositions.remove(event.pointer);
-    usedAndroidPointerIds.remove(pointerProperties[event.pointer]!.id);
-    pointerProperties.remove(event.pointer);
+  void _remove(int pointer) {
+    pointerPositions.remove(pointer);
+    usedAndroidPointerIds.remove(pointerProperties[pointer]!.id);
+    pointerProperties.remove(pointer);
     if (pointerProperties.isEmpty) {
       downTimeMillis = null;
     }
   }
 
+  void handlePointerUpEvent(PointerUpEvent event) {
+    _remove(event.pointer);
+  }
+
   void handlePointerCancelEvent(PointerCancelEvent event) {
-    pointerPositions.clear();
-    pointerProperties.clear();
-    usedAndroidPointerIds.clear();
-    downTimeMillis = null;
+    _remove(event.pointer);
```

Comment:
a comment would be helpful: "pointer cancel is handled like pointer up.  Normally, the difference is that pointer cancel doesn't perform any action, but in this case neither up or cancel perform any action".